pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -549,18 +549,13 @@ func (s *Reconciler) createNetworkInterface(ctx context.Context, nicName string)
 		Name:     nicName,
 		VnetName: azure.GenerateVnetName(s.scope.Cluster.Name),
 	}
-	switch set := s.scope.Machine.ObjectMeta.Labels[v1alpha1.MachineRoleLabel]; set {
-	case v1alpha1.Node:
-		networkInterfaceSpec.SubnetName = azure.GenerateNodeSubnetName(s.scope.Cluster.Name)
-	case v1alpha1.ControlPlane:
-		networkInterfaceSpec.SubnetName = azure.GenerateControlPlaneSubnetName(s.scope.Cluster.Name)
-		networkInterfaceSpec.PublicLoadBalancerName = azure.GeneratePublicLBName(s.scope.Cluster.Name)
```

Comment:
These names are must have, since control plane requires public and internal load balancer for access into the cluster.

its an unfortunate design hope to change it in future. The way it works now is

CreateControlPlaneVM -> CreateNetworkInterface -> AttachInterfacetoloadblancer

Instead hope to work this way

CreateControlPlaneVM -> CreateNetworkInterface
UpdateLoadBalancerRules with NetworkInterface

That way control plane is decoupled from load balancers.

Reply:
So you still have a use for the control plane code? How does the installer reuses the azure actuator these days?

Reply:
azure actuator only creates vm and sets up configuration of the base infra (example registering network interface with the load balancers)

as for worker nodes it just creates it on the base infra (right now all the names are hardcoded between installer and actuator hence little fragile), aim was to get it decoupled if we had time.

Reply:
Also azure will never have name conflicts so its ok to hardcode since all the resources are restricted to a certain resource group for each sub/tenantid